org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java

```diff
@@ -0,0 +1,66 @@
+package org.code.javabuilder;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.model.Message;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.amazonaws.services.sqs.model.ReceiveMessageResult;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class AWSInputAdapterTest {
+  private AWSInputAdapter inputAdapter;
+  private AmazonSQS sqsMock;
+
+  @BeforeEach
+  public void setUp() {
+    sqsMock = mock(AmazonSQS.class);
+    inputAdapter = new AWSInputAdapter(sqsMock, "url");
+  }
+
+  private void messageSetUp(String[] messages) {
```

Comment:
(minor) A header comment for this method might be useful.  Also, since you are setting precedents here, do you generally want to put helper functions before or after the tests?

Reply:
I'm used to putting them before because that's generally what our JS code does. Looking at our Ruby code, it has helpers more rarely, but when they are there, there seems to be a bit of a mix of before and after, possibly more after. Any reasons to favor one way over the other? I think I prefer before so I don't make any assumptions about built-in methods when I'm reading the code, but it's not a strong preference.

Reply:
I think I'm more used to seeing helper methods after the tests but don't know if that's a best practice or just a quirk of codebases I've worked on in the past.  As the person who's blazing the trail here, you get to choose!  :-)